elixir/apps/domain/lib/domain/actors/actor.ex

```diff
@@ -4,9 +4,7 @@ defmodule Domain.Actors.Actor do
   schema "actors" do
     field :type, Ecto.Enum, values: [:account_user, :account_admin_user, :service_account]
 
-    # TODO:
-    # field :first_name, :string
-    # field :last_name, :string
+    field :name, :string
```

Comment:
I would not keep them at all. Having optional fields that duplicate a required one is a source of confusion. Somebody will use them for query/searching or to render something in UI, for example, and then it will fail.

For example, they can not be used to "order by last name" because those fields are optional which means sometimes they will be empty. The search result in such a case makes no sense because we won't even use them for rendering results. Users can always sort by `name` and if the log is exported they can string split it (probably with bugs but good enough for audit cases) and do whatever they want with it.

Reply:
Hm, are you sure `name` is required and always there? The [spec mentions the `profile` scope is optional](https://openid.net/specs/openid-connect-basic-1_0.html#Scopes), and should return the following default profile claims:

`name, family_name, given_name, middle_name, nickname, preferred_username, profile, picture, website, gender, birthdate, zoneinfo, locale, and updated_at`

For example, the default Okta OIDC app mapping maps `displayName` to `name`:

<img width="995" alt="Screenshot 2023-06-27 at 8 26 25 AM" src="https://github.com/firezone/firezone/assets/167144/9055e792-50e4-4ac8-ba43-6cd84655c9d8">

However, adding a user in Okta doesn't have a field for `displayName` by default:

<img width="714" alt="Screenshot 2023-06-27 at 8 27 05 AM" src="https://github.com/firezone/firezone/assets/167144/fcb07018-4e70-418c-87bd-6c329c70aec0">

Which means `name` will be missing I believe.

Reply:
Or were you thinking we'd just concatenate `given_name` and `family_name` to populate the `name` field in our DB if needed?

Reply:
@jamilbk Of course, `name` is not always there but it will be available in more cases than other fields, because a lot of platforms don't ask for first/last names separately. People names are a hard problem that is very locale-specific, so I would try to avoid storing separate fields unless we do really need them. Also, syncing first/last names and joining them to a name is way easier than doing it in reverse.

Should we make it require though? It's a hard decision, from one point our UI will suffer a lot because you can't really tell who is who from user ids/group ids. We can't ask the user to enter their own name when IdP sync is enabled either, because then we will create sync issues (how to ignore certain fields) and possibly social engineering attack vectors (hey Mike can you please let me access X, it's your boss Brian).

BTW: We don't read it from claims directly but rather request from `userinfo` endpoint.

Reply:
> Or were you thinking we'd just concatenate given_name and family_name to populate the name field in our DB if needed?
Yes, if there is a `name` we use it, otherwise we concat.